src/main/java/picard/cmdline/argumentcollections/ReferenceArgumentCollection.java

```diff
@@ -25,13 +25,73 @@
 package picard.cmdline.argumentcollections;
 
 import java.io.File;
+import java.nio.file.Path;
+
+import htsjdk.samtools.util.Log;
+import picard.nio.PicardBucketUtils;
+import picard.nio.PicardHtsPath;
 
 /**
  * Base interface for a reference argument collection.
  */
 public interface ReferenceArgumentCollection {
     /**
-     * @return The reference provided by the user, if any, or the default, if any.
+     * This method is retained for backward compatibility with legacy tools that have not been updated to support PicardHtsPath input files.
+     * The preferred methods for accessing the reference file in the command line argument is either getHtsPath() or getReferencePath().
+     *
+     * @return The reference provided by the user, or the default defined by {@code htsjdk.samtools.Defaults.REFERENCE_FASTA}. May be null.
+     */
+    File getReferenceFile(); // tsato: update tools that call this method to convert File to PicardHtsPath and use getHtsPath()
+
+    /**
+     * This method first checks if the PicardHtsPath is null, thereby avoiding NPE that results from getHtsPath.toPath().
+     * Use this for providing input to methods that expect the Path to be null when the reference is absent e.g. SamReaderFactory.referenceSequence().
+     *
+     * @return The reference provided by the user or the default as an nio Path. May be null.
+     */
+    default Path getReferencePath(){
+        return getHtsPath() == null ? null : getHtsPath().toPath();
+    }
+
+    /**
+     * This default implementation is here to support tools that have not been upgraded to support cloud reference files.
+     * (The alternative is to make it abstract, which necessitates we update all the classes that implement this interface with
+     * an implementation like what we have below).
+     *
+     * The cloud-reference-enabled tools should override this method e.g. return REFERENCE_SEQUENCE (see DownsampleSam::makeReferenceArgumentCollection)
+     * Once all the relevant subclasses have been updated, we plan to make this abstract.
```

Comment:
I think this comment is somewhat out of place here and a bit confusing. Either way, I think the override cases are the exception, and should be (and I think are) documented where they occur. The javadoc here should really just describe the method itself. I would suggest the simplifying:
```suggestion
    /**
     * Returns a PicardHtsPath for the reference input. May be null. Implementers of this interface should override with an appropriate implementation.
```

Reply:
done